dev_guide/builds.adoc

```diff
@@ -354,3 +419,203 @@ If a build is triggered due to a webhook trigger or manual request,
+`Output` section of the `BuildConfig` and set it to the name of the `*Secret*`
+that you created, which in the above example is `*dockerhub*`:
+
+====
+----
+"parameters": {
+  "output": {
+    "to": {
+      "name": "private-image"
+    },
+    "pushSecret":{
+      "name":"dockerhub"
+    }
+  }
+}
+----
+====
+
+Pull the builder Docker image from a private Docker registry by specifying the
+`PullSecret` field, which is part of the build strategy definition:
+
+====
+----
+{
+  "strategy": {
+    "sourceStrategy": {
+      "from": {
+        "kind": "DockerImage",
+        "name": "docker.io/user/private_repository"
+       },
+       "pullSecret": {
+        "name": "dockerhub"
+       },
+    },
+    "type": "Source"
+  }
+}
+----
+====
+
+[#using-private-repositories-for-builds]
+== Using Private Repositories for Builds
+
+Supply valid credentials to build an application from a private repository.
+Currently, only SSH key based authentication is supported. The repository keys
+are located in the `$HOME/.ssh/` directory, and are named `id_dsa.pub`,
+`id_ecdsa.pub`, `id_ed25519.pub` or `id_rsa.pub` by default. Generate SSH key
+credentials with the following command:
+
+====
+
+----
+$ ssh-keygen -t rsa -C "your_email@example.com"
+----
+====
+
+Two files will be created: the public key (as explained above) and a
+corresponding private key (one of `id_dsa`, `id_ecdsa`, `id_ed25519` or
+`id_rsa`). With both of these in place you should consult your source control
+management (SCM) system's manual on how to upload the public key. The private
+key will be used to access your private repository.
+
+The
+https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/secrets.md[Secret]
+resource is used to store your keys. Create the `*Secret*` first before using
+the ssh key to access the private repository. The `*data*` field for the
+`*Secret*` object must contain your private key with the value set to the
+base64-encoded content of that file:
+
+====
+
+----
+$ base64 -w 0 $HOME/.ssh/id_rsa
+6yJodHRwc1ovL2zuZGV4LmRvY21lci5aby92MS8iOnsiYXV0aCI6ImJXWnZhblJwYXpwdVoybGxkR2d4TUE9PSIsImVtYWlsIj8ibWlAbWlmby5zayJ9fQ==
+----
+====
+
+Copy the value returned from the above command and place it into the
+`ssh-privatekey` field in `*_secret.json_*` file:
+
+====
+
+----
+{
+  "apiVersion": "v1beta3",
+  "kind": "Secret",
+  "metadata": {
+    "name": "scmsecret"
+  },
+  "data": {
+    "ssh-privatekey": "6yJodHRwc1ovL2zuZGV4LmRvY21lci5aby92MS8iOnsiYXV0aCI6ImJXWnZhblJwYXpwdVoybGxkR2d4TUE9PSIsImVtYWlsIj8ibWlAbWlmby5zayJ9fQ=="
+  }
+}
+
+----
+====
+
+Then, create the `*Secret*` from the *_secret.json_* file using the following
+command:
+
+====
+----
+$ oc create -f secret.json
+----
+====
+
+Add a `SourceSecret` field into the `Source` section inside the `BuildConfig`
+and set it to the name of the `*Secret*` that you created, in this case
+`*scmsecret*`:
+
+====
+
+----
+{
+  "apiVersion": "v1beta3",
```

Comment:
@soltysh @bparees this is kubernetes resource, do we have v1 for kubernetes as well?